src/charm.py

```diff
@@ -119,10 +119,11 @@ def api_port(self) -> int:
         return self._api_port
 
     @property
-    def peer_relation(self) -> Optional[Relation]:
+    def peer_relation(self) -> Relation:
```

Comment:
well technically this could still return `None`, and afaiu this would still qualify for `Optional`

Reply:
Correct. Totally agree, but in OF it's just Relation:
https://github.com/canonical/operator/blob/626527f988c007316218891830b1265fc936295e/ops/model.py#L137

```python
    def get_relation(
            self, relation_name: str,
            relation_id: typing.Optional[int] = None) -> 'Relation':
```
The question is what is correct and what should we do about it.
At the moment I think that OF should never return a None for a peer relation (type hint: Relation), but at the same time for other relation the type hint should be `Relation | None`.
This change here is a tradeoff. WDYT?